bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/framework/SystemBundleActivator.java

```diff
@@ -188,32 +191,25 @@ private void installSecurityManager(EquinoxConfiguration configuration) throws B
 	}
 
 	private void registerLocations(BundleContext bc, EquinoxLocations equinoxLocations) {
-		Dictionary<String, Object> locationProperties = new Hashtable<>(1);
-		Location location = equinoxLocations.getUserLocation();
-		if (location != null) {
-			locationProperties.put("type", EquinoxLocations.PROP_USER_AREA); //$NON-NLS-1$
-			register(bc, Location.class, location, locationProperties);
-		}
-		location = equinoxLocations.getInstanceLocation();
-		if (location != null) {
-			locationProperties.put("type", EquinoxLocations.PROP_INSTANCE_AREA); //$NON-NLS-1$
-			register(bc, Location.class, location, locationProperties);
-		}
-		location = equinoxLocations.getConfigurationLocation();
-		if (location != null) {
-			locationProperties.put("type", EquinoxLocations.PROP_CONFIG_AREA); //$NON-NLS-1$
-			register(bc, Location.class, location, locationProperties);
-		}
-		location = equinoxLocations.getInstallLocation();
-		if (location != null) {
-			locationProperties.put("type", EquinoxLocations.PROP_INSTALL_AREA); //$NON-NLS-1$
-			register(bc, Location.class, location, locationProperties);
-		}
+		registerLocation(bc, equinoxLocations.getUserLocation(), EquinoxLocations.PROP_USER_AREA);
+		registerLocation(bc, equinoxLocations.getInstanceLocation(), EquinoxLocations.PROP_INSTANCE_AREA);
+		registerLocation(bc, equinoxLocations.getConfigurationLocation(), EquinoxLocations.PROP_CONFIG_AREA);
+		registerLocation(bc, equinoxLocations.getInstallLocation(), EquinoxLocations.PROP_INSTALL_AREA);
+		registerLocation(bc, equinoxLocations.getEclipseHomeLocation(), EquinoxLocations.PROP_HOME_LOCATION_AREA);
+	}
 
-		location = equinoxLocations.getEclipseHomeLocation();
+	private void registerLocation(BundleContext bc, BasicLocation location, String type) {
 		if (location != null) {
-			locationProperties.put("type", EquinoxLocations.PROP_HOME_LOCATION_AREA); //$NON-NLS-1$
-			register(bc, Location.class, location, locationProperties);
+			Dictionary<String, Object> locationProperties = new Hashtable<>(1);
+			locationProperties.put("type", type); //$NON-NLS-1$
+			URL url = location.getUrlOrDefault();
```

Comment:
Should we have a separate property for the default URL?

Here you use url.toString while in BasicLocation you use toExternalForm.  I suggest we use toExternalForm to be consistent even though toString just calls toExternalForm.

May want to consider just placing the URL in directly as the value.

Reply:
I can add ur/default url separately, about the URL as an object I'm always trying to stick to the types supported by DS, is URL a type supported by DS/OSGi that could be processed in a special way (e.g. something like property.url.host=....)?

Reply:
I added the default url now